dev_test_cmd.sh

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env bash
+
+# Catch any non-flagged use
+if [ $# -eq 0 ]; then
+    echo "Invalid: $(basename "$0") has required flags, use: -h for help"
+    exit 1
+fi
+
+# e: Stops execution in the instance of a command or pipeline error
+# u: Treat unset variables as an error and exit immediately
+set -eu
+
+# Default to FAILED
+STATUS="FAILED"
+
+function outcome_and_exit {
+  if [ $STATUS == "SUCCESS" ]
+  then
+    echo -e "\e[32m$STATUS\e[0m"
+  else
+    echo -e "\e[31m$STATUS\e[0m"
+  fi
+}
+
+while getopts ":ha" opt; do # Parse options to the `test_cmd` command
+  case ${opt} in
+    h )
+      echo
+      echo "Usage:"
+      echo "    -h             Displays this help message"
+      echo "    -a             Run all unit and integration tests"
+      echo
+      echo "Commands:"
+      echo "    unit           Unit tests subcommand, run unit -h for more information"
+      echo "    integration    Integration tests subcommand, run integration -h for more information"
+      exit 0
+      ;;
+    a ) # -a (all)
+      docker-compose run --use-aliases boulder ./test.sh
+      ;;
+   \? ) # catch invalid options
+      echo "Invalid Option: $OPTARG use: -h for help" 1>&2
+      exit 1
+      ;;
+  esac
+done
+shift $((OPTIND -1))
+subcommand=$1; shift
+
+# Catch any non-flagged subcommand usage
+if [ $# -eq 0 ]; then
+    echo "Invalid: $(basename "$0") <COMMAND> has required flags, use: -h for help"
+    exit 1
+fi
+
+case "$subcommand" in
+  unit) # Parse options to the unit sub command
+    while getopts ":d:ha" opt; do
+      case ${opt} in
+        h ) # -h (help)
+          echo
+          echo "Usage:"
+          echo "    -h                Displays this help message"
+          echo "    -a                Run all unit tests"
+          echo "    -d <DIRECTORY>    Run unit tests for a specific directory"
+          STATUS="HELP"
+          exit 0
+          ;;
+        a )  # -a (all)
+          trap "outcome_and_exit" EXIT 
```

Comment:
I think outcome_and_exit can actually run once at the top level.

BTW, we should remove the SUCCESS / FAILURE prints from integration-test.py so we don't wind up printing the status twice.

Reply:
> BTW, we should remove the SUCCESS / FAILURE prints from integration-test.py so we don't wind up printing the status twice.

The addition of SUCCESS (green) / FAILURE (red) is born out of my own confusion when parsing the integration test output. Given the 47 lines of SIGTERM messages the outcome was not very prominent and depending on font and terminal window size an operator may have trouble finding it.

> I think outcome_and_exit can actually run once at the top level.

Given the above, if I were to keep the test outcome messages, there would need to be additional logic to not print FAILURE on exit for missing flags or invalid options. I briefly had written a version like this and this ended up looking cleaner to my eye. If we remove the test outcome messages we can definitely just include this once.

Reply:
> BTW, we should remove the SUCCESS / FAILURE prints from integration-test.py so we don't wind up printing the status twice.
Oh gosh, I think I just realized what you mean here. I had, for some reason, assumed this was in reference to the wrapper calling it. Yeah, I'm happy to remove that line from `integration-test.py`. As long as our Travic CI code doesn't depend on seeing `SUCCESS` when calling `test.sh` in it's current from.

Reply:
Yep, I meant take it out of the .py because we put it in the .sh. I totally agree with you that the SUCCESS/FAILURE printed by the .py gets lost in the noise, and we should fix that.

By the way, the red messages like Failed to generate OCSP response and something something unary, we should fix, since they're distracting. They're a little challenging to do, though; let's discuss in another issue.